src/lib/utils.ts

```diff
@@ -1,18 +1,213 @@
-import { Context } from "telegraf";
-import { EBotUserState, IBotUser } from "../interfaces/bot-users";
+import { Context } from 'telegraf';
+import { EBotUserState, IBotUser } from '../interfaces/bot-users';
+import axios from 'axios';
+import { v4 as uuidv4 } from 'uuid';
+
+const BOT_API_KEY = process.env.BOT_API_KEY;
+const API_HOST_URL = process.env.API_HOST_URL;
+const DYNAMIC_TOKEN = process.env.DYNAMIC_TOKEN;
+const DYNAMIC_ENVIRONMENT_ID = process.env.DYNAMIC_ENVIRONMENT_ID;
+
+const headers = {
+  Authorization: `Bearer ${DYNAMIC_TOKEN}`,
+  'Content-Type': 'application/json',
+};
+const baseUrl = `https://app.dynamicauth.com/api/v0`;
 
 export const adaptCtx2User = (ctx: Context) => {
-    const newUser: IBotUser = {
-        tg_id: ctx.from?.id!,
-        tg_first_name: ctx.from?.first_name!,
-        tg_username: ctx.from?.username!,
-        tg_is_bot: ctx.from?.is_bot!,
-        tg_language_code: ctx.from?.language_code!,
-        state: EBotUserState.NEW,
-        original_email_address: "overwrite",
-        // TODO use chomp API to get an ID
-        // chomp_id: "TODO",
-    }
-
-    return newUser
-}
\ No newline at end of file
+  const newUser: IBotUser = {
+    tg_id: ctx.from?.id!,
+    tg_first_name: ctx.from?.first_name!,
+    tg_username: ctx.from?.username!,
+    tg_is_bot: ctx.from?.is_bot!,
+    tg_language_code: ctx.from?.language_code!,
+    state: EBotUserState.NEW,
+    original_email_address: 'overwrite',
+  };
+
+  return newUser;
+};
+
+// Create temporary user with random uuid using Telegram ID
+export const createUserByTelegram = async (tgId: number) => {
```

Comment:
Can you cast this to a specific custom type like `IChompUser` so its clear what fields are available